plugins/ptp_operator/event/event.go

```diff
@@ -66,14 +69,43 @@ type PMetric struct {
 	metricCounter *prometheus.Counter
 }
 
+func (dt *DependingClockState) hasClockState(iface string) (int, *ClockState) {
+	for index, dep := range *dt {
+		if *dep.IFace == iface {
+			return index, dep
+		}
+	}
+	return -1, nil
+}
+
+func (dt *DependingClockState) hasMetric() map[string]*PMetric {
```

Comment:
shouldn't this return Metric based on specific iface similar to `hasClockState`?

Reply:
it is based on dependent type, metrics is common for all interface based on type,  so you are checking is there is metrics object already define. You can't register same metrics twice